### PR TITLE
fix(fluent-button): fix typo in compound disabled selector, add tests

### PR DESCRIPTION
**Pull request by davatron5000**
## Previous Behavior

Found a syntax issue in compound disabled CSS selector for `<fluent-button>` and `<fluent-anchor-button>`. No existing behavior because CSS was invalid.

## New Behavior

Adding the `disabled` attribute and the `disabled-focusable` attribute to a FluentButton or FluentAnchorButton with `[appearance='primary']` will have a transparent border. 

## Related Issue(s)

- Fixes #

### CODE REVIEW

File: packages/web-components/src/anchor-button/anchor-button.spec.ts [hunk 1]
```diff
@@ -103,4 +103,22 @@ test.describe('Anchor Button - Boolean Attributes', () => {
       await expect(element).toHaveJSProperty(attribute, value);
     });
   }
+
+  test(`should have transparent border when the \`disabled\` or \`disabled-focus\` attribute is present`, async ({
+    page,
+  }) => {
+    await element.evaluate((node: any) => {
+      node.getAttribute('appearance', 'primary');
```
Comment (radium-v) [inline packages/web-components/src/anchor-button/anchor-button.spec.ts:111]: Did you mean to set this attribute?

```suggestion
      node.setAttribute('appearance', 'primary');
```

Comment (davatron5000) [reply packages/web-components/src/anchor-button/anchor-button.spec.ts]: I did mean that! Fixed.

**Review by chrisdholt — APPROVED** [review]
Will leave to @radium-v to sign off and manage with the current merge happening.

**Review by radium-v — APPROVED** [review]

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

File: change/@fluentui-web-components-f04748f8-f9ea-46fa-a74b-b301279eb3c3.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-web-components-f04748f8-f9ea-46fa-a74b-b301279eb3c3.json:1]: ## [🕵 fluentui-web-components-v3 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=31302&commitId=85a963fc0d57b20ecea36d17fac77b74d92df8a4&env=prod&clientType=FLUENTUI)


  <details><summary><b>Badge</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Badge.Appearance - Dark Mode.normal.chromium.png|15| Changed|

</details>


  <details><summary><b>Slider</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Slider.Slider Steps.normal.chromium.png|1| Changed|

</details>


  <details><summary><b>Text</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Text.Block.normal.chromium.png|995| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_web-components"/>

Comment (davatron5000) [reply change/@fluentui-web-components-f04748f8-f9ea-46fa-a74b-b301279eb3c3.json:1]: I don't think these are related to changes. Due to Shadow DOM encapsulation I wouldn't expect other components to be effected by these changes.

- [x] Hexes for this branch against what's in the production Storybook ([Badge w/ theme set to Teams Dark](https://web-components.fluentui.dev/?path=/docs/components-badge-badge--badge#appearance)) for the background of the "tint" variant is the same: `#2f2f4a`
- [x] Slider change appears to be compression noise
- [x] Light theme Text Blocks appear the same on localhost as on prod. 

I've marked these as noise. But let me know if I did something wrong, this is my first PR.

Comment (chrisdholt) [reply change/@fluentui-web-components-f04748f8-f9ea-46fa-a74b-b301279eb3c3.json:1]: Just noise :)

**Review by janechu — APPROVED** [review]

**Review by chrisdholt — APPROVED** [review]

**Pull request merged**